numpy/lib/function_base.py

```diff
@@ -1455,26 +1455,31 @@ def angle(z, deg=False):
     return a
 
 
-def _unwrap_dispatcher(p, discont=None, axis=None):
+def _unwrap_dispatcher(p, discont=None, axis=None, *, min_val=None, max_val=None):
     return (p,)
 
 
 @array_function_dispatch(_unwrap_dispatcher)
-def unwrap(p, discont=pi, axis=-1):
+def unwrap(p, discont=None, axis=-1, *, min_val=-pi, max_val=pi):
     """
-    Unwrap by changing deltas between values to 2*pi complement.
-
-    Unwrap radian phase `p` by changing absolute jumps greater than
-    `discont` to their 2*pi complement along the given axis.
+    Unwrap by changing deltas between values to complement.
+    
+    For `min_val=-pi` and `max_val=pi`, unwraps radian phase `p` by 
+    changing absolute jumps greater than `discont` to their 2*pi 
+    complement along the given axis.
 
     Parameters
     ----------
     p : array_like
         Input array.
     discont : float, optional
-        Maximum discontinuity between values, default is ``pi``.
+        Maximum discontinuity between values, default is ``(max_val - min_val)/2``.
     axis : int, optional
         Axis along which unwrap will operate, default is the last axis.
+    min_val : float, optional
+        Minimum value when sequence overflows, default is ``-pi``
+    max_val : float, optional
+        Maximum value when sequence underflows, default is ``pi``
 
     Returns
     -------
```

Comment:
The notes section below still has a hard-coded `pi`, so probably needs rewording to be more general

Reply:
While the value in the function is `None` the effective default value is `pi` so it works on existing cases, did you want to change this?

Reply:
At the very least I'd like the notes section to include a sentence like "for the default case when ..., ...". But if you can rewrite it in terms of `min_val`, `max_val`, and `discont`, that would be more useful.